sipa/blueprints/usersuite.py

```diff
@@ -613,3 +616,89 @@ def reset_wifi_password():
     return render_template('generic_form.html',
                            page_title=gettext("Neues WLAN Passwort"),
                            form_args=form_args)
+
+
+@bp_usersuite.route("/request-repayment", methods=["GET", "POST"])
+@login_required
+def request_repayment():
+    """
+    Request a repayment of excess membership contributions
+    """
+
+    form = RequestRepaymentForm()
+
+    if form.validate_on_submit():
+        return redirect(
+            url_for(
+                ".request_repayment_confirm",
+                beneficiary=form.beneficiary.data,
+                iban=form.iban.data,
+                amount=form.amount.data,
+            )
+        )
+    elif form.is_submitted():
+        flash_formerrors(form)
+
+    form_args = {
+        "form": form,
+        "cancel_to": url_for(".index"),
+        "submit_text": gettext("Weiter"),
+    }
+
+    return render_template(
+        "generic_form.html",
+        page_title=gettext("Rücküberweisung beantragen"),
+        form_args=form_args,
+    )
+
+
+@bp_usersuite.route("/request-repayment/confirm", methods=["GET", "POST"])
+@login_required
+def request_repayment_confirm():
+    """
+    Request a repayment of excess membership contributions
+    """
+
+    beneficiary = request.args.get("beneficiary", None)
+    iban = request.args.get("iban", None, lambda x: IBAN(x, validate_bban=True))
+    amount = request.args.get("amount", None, lambda x: Decimal(x))
+
+    form = RequestRepaymentConfirmForm()
```

Comment:
I'm confused about the intention here. Why are you parsing query-arguments via [`requests.args`](https://flask.palletsprojects.com/en/2.3.x/api/#flask.Request.args), but at the same time have form data as given by this form?

Reply:
First, the URL flow is as follows: `usersuite -> usersuite/request-repayment -> usersuite/request-repayment/confirm -> usersuite` .

After POST submitting the form to `usersuite/request-repayment`, a redirect (resulting in a GET) to the `usersuite/request-repayment/confirm` confirmation page should happen.  The redirect URL has the data from the POSTed form in the query parameters.

> Why are you parsing query-arguments via [requests.args](https://flask.palletsprojects.com/en/2.3.x/api/#flask.Request.args), but at the same time have form data as given by this form?

I thought this is necessary in order to get the data from the query parameters.  When receiving the GET, the form is still empty, it does not contain data before the final submission POST.

Maybe there is an easier way to do achieve this workflow, I just could not think of any. :)

Reply:
So is the extra step asking for any more information than a „revisit your data and confirm it“?
I think we do something similar somewhere else, but with a hidden checkbox. Grep in sipa or pycroft, you will find it.

Reply:
> So is the extra step asking for any more information than a „revisit your data and confirm it“?

No, just like in the registration part.  Resolving as discussed offline.